File: frontend/src/types/analysis.ts
export interface Recommendation {
    title: string;
    message: string;
    type: string;
}

export interface AnalysisResult {
    match_score: number;
    ats_score: number;
    grade: string;
    summary: string;
    matched_skills: string[];
    missing_skills: string[];
    resume_skills: string[];
    required_skills: string[];
    recommendations: Recommendation[];
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel


class AnalysisRequest(BaseModel):
    resume_text: str
    job_description: str


class Recommendation(BaseModel):
    title: str
    message: str
    type: str


class AnalysisResponse(BaseModel):
    analysis_id: str
    match_score: float
    ats_score: int
    grade: str
    summary: str
    matched_skills: list[str]
    missing_skills: list[str]
    resume_skills: list[str]
    required_skills: list[str]
    recommendations: list[Recommendation]
